tests/api2/test_425_smb_protocol.py

```diff
@@ -275,35 +281,81 @@ def test_061_create_and_write_stream_smb2(request):
 
     fd4 = c.create_file("streamstestdir:smb2_stream", "w")
     contents2 = c.read(fd4, 0, 5)
     c.close(fd4)
+    kcontent2 = get_stream('streamstestdir', 'smb2_stream')
 
     c.rmdir("streamstestdir")
     c.disconnect()
     assert (contents.decode() == "test1")
     assert (contents2.decode() == "test2")
 
+    # Remove samba compatibility NULL byte
+    assert kcontent1[:-1].decode() == 'test1'
+    assert kcontent2[:-1].decode() == 'test2'
+
 
 @pytest.mark.dependency(name="LARGE_STREAM_WRITTEN_SMB2")
-def test_062_write_stream_large_offset_smb2(request):
+def test_062_write_stream_large_offset_smb2(request, mount_share):
     """
     Append to our existing stream over SMB2/3 protocol. Specify an offset that will
     cause resuling xattr to exceed 64KiB default xattr size limit in Linux.
     """
     depends(request, ["STREAM_TESTFILE_CREATED"])
-    c = SMB()
-    c.connect(host=ip, share=SMB_NAME, username=SMB_USER, password=SMB_PWD, smb1=False)
-    fd = c.create_file("streamstestfile:smb2_stream", "w")
-    c.write(fd, b'test2', 131072)
-    c.close(fd)
+    with smb_connection(
+        host=ip,
+        share=SMB_NAME,
+        username=SMB_USER,
+        password=SMB_PWD,
+        smb1=False
+    ) as c:
+        fd = c.create_file("streamstestfile:smb2_stream", "w")
+        try:
+            c.write(fd, b'test2', 131072)
+        finally:
+            c.close(fd)
 
-    fd2 = c.create_file("streamstestfile:smb2_stream", "w")
-    contents = c.read(fd2, 131072, 5)
-    c.close(fd2)
-    c.disconnect()
-    assert (contents.decode() == "test2")
+        fd2 = c.create_file("streamstestfile:smb2_stream", "w")
+        try:
+            contents = c.read(fd2, 131072, 5)
+        finally:
+            c.close(fd2)
+
+        kcontent = get_stream('streamstestfile', 'smb2_stream')
+
+        assert (contents.decode() == "test2")
+
+        # Verify that reading a large stream functions correctly
+        assert len(kcontent) == 131072 + 5 + 1
+
+        # Remove samba compatibility NULL byte
+        assert kcontent[131072:-1].decode() == 'test2'
+
+        # Verify that SMB server rejects too-large stream write
+        fd = c.create_file("streamstestfile:smb2_stream", "w")
+        try:
+            with pytest.raises(NTSTATUSError) as e:
+                c.write(fd, b'test2', 2097152)
+
+            assert e.value.args[0] == ntstatus.NT_STATUS_FILE_SYSTEM_LIMITATION
+        finally:
+            c.close(fd)
+
+        # Verify that SMB server allows _very_ large write
+        fd = c.create_file("streamstestfile:smb2_stream", "w")
+        try:
+            # We have to an extra byte for that nul at end of xattr
+            offset = 2097152 - (len(b"test2") + 1)
+            c.write(fd, b"test2", offset)
+            contents = c.read(fd, offset, 5)
+            assert contents.decode() == "test2"
+        finally:
+            c.close(fd)
+
+        copy_stream('streamstestfile', 'smb2_stream', 'smb2_stream2')
```

Comment:
After the `copy_stream` seems like a perfect opportunity to:
1. Run `list_stream` and ensure that the resultant set is as expected.
2. Read _smb2_stream2_ using _**both**_ SMB and `get_stream` and validate contents.